swri_transform_util/src/swri_transform_util/wgs84_transformer.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+
+"""A module containing a simple class for transforming from WGS84 coordinate
+frame to a local_xy frame and vice versa."""
+
+# Copyright (C) 2020, Southwest Research Institute® (SwRI®)
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import math
+import numpy as np
+from tf.transformations import euler_from_quaternion
+
+EARTH_ECCENTRICITY = 0.08181919084261
+EARTH_EQUATOR_RADIUS = 6378137.0
+
+
+class Wgs84Transformer(object):
+    """
+    A simple class for transforming from the WGS84 frame to a local_xy frame 
+    and vice versa.
+
+    Transforms to and from WGS84 and the local map frame using a local origin
+    as a reference.
+    """
+
+    def __init__(self, local_origin):
+        """
+        Constructor for the Wgs84Transformer class
+        :param geometry_msgs.Pose local_origin: An initialized local origin
+        """
+        self._reference_heading = -1.0 * euler_from_quaternion(quaternion=(local_origin.pose.orientation.x,
+                                                                           local_origin.pose.orientation.y,
+                                                                           local_origin.pose.orientation.z,
+                                                                           local_origin.pose.orientation.w))[2]  # get yaw from quaternion
```

Comment:
PEP-8 requires lines be < 100 characters long.
```suggestion
        self._reference_heading = -1.0 * euler_from_quaternion(
            quaternion=(local_origin.pose.orientation.x,
                        local_origin.pose.orientation.y,
                        local_origin.pose.orientation.z,
                        local_origin.pose.orientation.w))[2]  # get yaw from quaternion
```